GGRC-1043 Fix corrupted order of revisions

pod2metra opened the pull request:
change the order of revision log creation 
create should be at first 
delete - at last 
modified  - in the middle

zidarsk8 commented:
@pod2metra @alieh-rymasheuski 
If I understand this correctly, the wrong order for an object can only happen if the same object is modified and deleted in the same request.

This seems like a good fix, but I'm a bit worried on how that can even happen.
Do we know what are the steps needed to produce this bug, so we can actually verify it?

pod2metra commented:
@zidarsk8 

>>> cached_objects = {"modified": [1,2,3], "deleted": [4,5,6], "created": [7,8,9], }
>>> cached_objects.items()
[('deleted', [4, 5, 6]), ('modified', [1, 2, 3]), ('created', [7, 8, 9])]

for your note if in cached objects there are same object in deleted and modified and created then in revisions will be first delete revision after that modified and only after that deleted 
if you have in same moment deleted and modified object then you will have deleted at first and modified after. 
if you have created and modified then at first will be modified and at last will be created. 
This is the problem №1

The problem №2: 
If you have ownable object in pull its owner in created pull too then you will have modified before created on owner instance

alieh-rymasheuski commented:
@zidarsk8 Technically, there is no way to log two revisions for the same resource in one request except for Ownable objects. When you update an Ownable object together with its owner list (like, you create or delete it), you have an unpredictable order of these two things resolving:
1. A create/delete revision is logged for the resource.
2. An update revision is logged for the resource due to the changed owners list.

The order depends on the order of a `for key in {"created": [], "updated": [], "deleted": []}` (we had the same kind of error in JSON builder some time ago).
